src/main/java/com/google/api/codegen/transformer/csharp/CSharpSurfaceNamer.java

```diff
@@ -30,16 +30,78 @@
 import com.google.api.tools.framework.model.Interface;
 import com.google.api.tools.framework.model.Method;
 import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.base.Function;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import java.util.List;
+import java.util.Map;
 
 public class CSharpSurfaceNamer extends SurfaceNamer {
 
+  public static final class MethodFields {
+    public MethodFields(Method method, List<Field> fields) {
```

Comment:
It uses a deep equality of the field list, so requires a custom hashcode and equals. I expect this is overridable in AutoValue, but it didn't really seem worth it.